### PR TITLE
SCMOD-9609: Adding elasticsearch credentials

### PR DESCRIPTION
**Pull request by rorytorneymf**
Jira: https://portal.digitalsafe.net/browse/SCMOD-9609

### CODE REVIEW

File: caf-audit-monkey/README.md [hunk 1]
```diff
@@ -103,6 +103,16 @@ The following parameters may be set as required:
     <td>NONE, 9200</td>
     <td>The REST port of the Elasticsearch server listens on. eg. 9200. This is an alternative variable used to construct elastic search host and port by combining with CAF_ELASTIC_HOST_VALUES.</td>
   </tr>
+  <tr>
+    <td>CAF_ELASTIC_USERNAME</td>
+    <td>null, [Any String]</td>
+    <td>ElasticSearch username. Defaults to null (anonymous access).</td>
```
Comment (dermot-hardy) [inline caf-audit-monkey/README.md]: Elasticsearch is one word.

```suggestion
    <td>Elasticsearch username. Defaults to null (anonymous access).</td>
```

File: caf-audit-monkey/README.md [hunk 2]
```diff
@@ -103,6 +103,16 @@ The following parameters may be set as required:
...
+  </tr>
+  <tr>
+    <td>CAF_ELASTIC_PASSWORD</td>
+    <td>null, [Any String]</td>
+    <td>ElasticSearch password. Defaults to null (anonymous access).</td>
```
Comment (dermot-hardy) [inline caf-audit-monkey/README.md]:
```suggestion
    <td>Elasticsearch password. Defaults to null (anonymous access).</td>
```

File: caf-audit/README.md [hunk 3]
```diff
@@ -155,6 +155,10 @@ Configuration required to be supplied via environment variable:
 
 - `CAF_ELASTIC_PORT_VALUE`: The REST port of the ElasticSearch server listens on. e.g. 9200
 
+- `CAF_ELASTIC_USERNAME`: ElasticSearch username. Defaults to null (anonymous access).
```
Comment (dermot-hardy) [inline caf-audit/README.md]:
```suggestion
- `CAF_ELASTIC_USERNAME`: Elasticsearch username. Defaults to null (anonymous access).
```

File: caf-audit/README.md [hunk 4]
```diff
@@ -155,6 +155,10 @@ Configuration required to be supplied via environment variable:
...
+
+- `CAF_ELASTIC_PASSWORD`: ElasticSearch password. Defaults to null (anonymous access).
```
Comment (dermot-hardy) [inline caf-audit/README.md]:
```suggestion
- `CAF_ELASTIC_PASSWORD`: Elasticsearch password. Defaults to null (anonymous access).
```

File: caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditRestHighLevelClientFactory.java [hunk 5]
```diff
@@ -45,10 +49,13 @@ private ElasticAuditRestHighLevelClientFactory() {
      * Returns an elastic search high level client.
      *
      * @param hostAndPortValues comma separated list of Elasticsearch host:port values
+     * @param elasticUsername comma ElasticSearch username
+     * @param elasticPassword comma ElasticSearch password
```
Comment (dermot-hardy) [inline caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditRestHighLevelClientFactory.java]:
```suggestion
     * @param elasticUsername Elasticsearch username
     * @param elasticPassword Elasticsearch password
```

File: deploy/src/main/resources/production-marathon/README.md [hunk 6]
```diff
@@ -22,6 +22,10 @@ The `environment.sh` file supports configurable property settings necessary for
 
 - `CAF_ELASTIC_PORT_VALUE`: The REST port of the ElasticSearch server listens on. e.g. 9200
 
+- `CAF_ELASTIC_USERNAME`: ElasticSearch username. Defaults to null (anonymous access).
+
+- `CAF_ELASTIC_PASSWORD`: ElasticSearch password. Defaults to null (anonymous access).
```
Comment (dermot-hardy) [inline deploy/src/main/resources/production-marathon/README.md]:
```suggestion
- `CAF_ELASTIC_USERNAME`: Elasticsearch username. Defaults to null (anonymous access).

- `CAF_ELASTIC_PASSWORD`: Elasticsearch password. Defaults to null (anonymous access).
```

File: deploy/src/main/resources/production-marathon/environment.sh [hunk 7]
```diff
@@ -11,3 +11,9 @@ export CAF_ELASTIC_HOST_VALUES=
 
 # The REST port of the ElasticSearch server listens on.
 export CAF_ELASTIC_PORT_VALUE=
+
+# ElasticSearch username.
+export CAF_ELASTIC_USERNAME=
+
+# ElasticSearch password.
```
Comment (dermot-hardy) [inline deploy/src/main/resources/production-marathon/environment.sh]:
```suggestion
# Elasticsearch username.
export CAF_ELASTIC_USERNAME=

# Elasticsearch password.
```

File: deploy/src/main/resources/production-marathon/production-prereqs/README.md [hunk 8]
```diff
@@ -20,6 +20,10 @@ The `environment-testing.sh` file supports configurable property settings necess
 
 - `CAF_TESTING_ELASTICSEARCH_CLUSTER_NAME`: This configures the name of the Elasticsearch cluster. e.g. audit-smoketest-elasticsearch.
 
+- `CAF_ELASTIC_USERNAME`: ElasticSearch username. Defaults to null (anonymous access).
+
+- `CAF_ELASTIC_PASSWORD`: ElasticSearch password. Defaults to null (anonymous access).
```
Comment (dermot-hardy) [inline deploy/src/main/resources/production-marathon/production-prereqs/README.md]:
```suggestion
- `CAF_ELASTIC_USERNAME`: Elasticsearch username. Defaults to null (anonymous access).

- `CAF_ELASTIC_PASSWORD`: Elasticsearch password. Defaults to null (anonymous access).
```

File: deploy/src/main/resources/production-swarm/README.md [hunk 9]
```diff
@@ -29,6 +29,10 @@ The `environment.sh` file supports configurable property settings necessary for
 	export CAF_ELASTIC_NUMBER_OF_REPLICAS=1
 	## The REST port of the Elasticsearch server listens on.
 	export CAF_ELASTIC_PORT_VALUE=9200
+	## ElasticSearch username.
+	export CAF_ELASTIC_USERNAME=
+	## ElasticSearch password.
```
Comment (dermot-hardy) [inline deploy/src/main/resources/production-swarm/README.md]:
```suggestion
	## Elasticsearch username.
	export CAF_ELASTIC_USERNAME=
	## Elasticsearch password.
```

File: deploy/src/main/resources/production-swarm/environment.sh [hunk 10]
```diff
@@ -18,3 +18,7 @@ export CAF_ELASTIC_NUMBER_OF_SHARDS=5
 export CAF_ELASTIC_NUMBER_OF_REPLICAS=1
 ## The REST port of the Elasticsearch server listens on.
 export CAF_ELASTIC_PORT_VALUE=9200
+## ElasticSearch username.
+export CAF_ELASTIC_USERNAME=
+## ElasticSearch password.
```
Comment (dermot-hardy) [inline deploy/src/main/resources/production-swarm/environment.sh]:
```suggestion
## Elasticsearch username.
export CAF_ELASTIC_USERNAME=
## Elasticsearch password.
```

File: deploy/src/main/resources-filtered/production-marathon/marathon.json.b [hunk 11]
```diff
@@ -26,7 +26,9 @@
             "CAF_ELASTIC_NUMBER_OF_REPLICAS": "1",
             "CAF_ELASTIC_NUMBER_OF_SHARDS": "5",
             "CAF_ELASTIC_PORT_VALUE": "${CAF_ELASTIC_PORT_VALUE}",
-            "CAF_LOG_LEVEL": "INFO"            
+            "CAF_ELASTIC_USERNAME": "${CAF_ELASTIC_USERNAME}",
+            "CAF_ELASTIC_PASSWORD": "${CAF_ELASTIC_PASSWORD}",
+            "CAF_LOG_LEVEL": "INFO",
```
Comment (dermot-hardy) [inline deploy/src/main/resources-filtered/production-marathon/marathon.json.b]: Are you allowed trailing commas?  There's no reason to change this line anyway.

```suggestion
            "CAF_LOG_LEVEL": "INFO"            
```

File: docs/pages/en-us/Client-API.md [hunk 12]
```diff
@@ -29,6 +29,8 @@ Configuration required to be supplied via environment variable:
 - `CAF_ELASTIC_PROTOCOL`: The protocol used to connect to the Elasticsearch server. e.g. http or https. Default value is http.
 - `CAF_ELASTIC_NUMBER_OF_SHARDS`: The number of shards elasticsearch is configured to use.  
 - `CAF_ELASTIC_NUMBER_OF_REPLICAS`: The number of replicas configured for elasticsearch.
+- `CAF_ELASTIC_USERNAME`: ElasticSearch username. Defaults to null (anonymous access).
+- `CAF_ELASTIC_PASSWORD`: ElasticSearch password. Defaults to null (anonymous access).
```
Comment (dermot-hardy) [inline docs/pages/en-us/Client-API.md]:
```suggestion
- `CAF_ELASTIC_USERNAME`: Elasticsearch username. Defaults to null (anonymous access).
- `CAF_ELASTIC_PASSWORD`: Elasticsearch password. Defaults to null (anonymous access).
```

File: docs/pages/en-us/Getting-Started.md [hunk 13]
```diff
@@ -263,6 +263,8 @@ The `AuditConnection` object, for direct to Elasticseach, can be constructed by
         - `CAF_ELASTIC_NUMBER_OF_REPLICAS` the number of replica shards (copies) that each primary shard should have. Defaults to 1.
         - `CAF_ELASTIC_HOST_VALUES` a comma separated list of hostnames to use when contacting elasticsearch. eg. localhost, otherHost.
         - `CAF_ELASTIC_PORT_VALUE` the REST port of the ElasticSearch server listens on. e.g. 9200.
+        - `CAF_ELASTIC_USERNAME`: ElasticSearch username. Defaults to null (anonymous access).
+        - `CAF_ELASTIC_PASSWORD`: ElasticSearch password. Defaults to null (anonymous access).
```
Comment (dermot-hardy) [inline docs/pages/en-us/Getting-Started.md]:
```suggestion
        - `CAF_ELASTIC_USERNAME`: Elasticsearch username. Defaults to null (anonymous access).
        - `CAF_ELASTIC_PASSWORD`: Elasticsearch password. Defaults to null (anonymous access).
```

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

**Comment by andyreidz** [comment]
@dermot-hardy The suggested changes feature is nice. 
Shame that multiple changes to a single file can't be batched and applying a change forces a page reload. It makes applying the changes take much longer and also collapses the hidden conversations each time.

**Review by dermot-hardy — APPROVED** [review]
If you really wanted to you could do another PR into this one for that.

**Pull request merged**